crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -387,21 +379,40 @@ impl CertifiedCheckpointSummary {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct CheckpointProposalContents {
+    // TODO: Currently we are not really using the effects digests, but in the future we may be
+    // able to use it to optimize the sync process.
+    pub transactions: BTreeSet<ExecutionDigests>,
+}
+
+impl CheckpointProposalContents {
+    pub fn new<T>(contents: T) -> Self
+    where
+        T: Iterator<Item = ExecutionDigests>,
+    {
+        Self {
+            transactions: contents.collect(),
+        }
+    }
+
+    pub fn digest(&self) -> CheckpointContentsDigest {
+        sha3_hash(self)
+    }
+}
+
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct CheckpointContents {
     pub transactions: Vec<ExecutionDigests>,
```

Comment:
can you add a doc comment clarifying that transactions is always assumed to be causally ordered?